src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp

```diff
@@ -111,25 +111,20 @@ ReportPlatform::getPropertyTree20202( const xrt_core::device * device,
   pt_platform.put("hardware.serial_num", info.mSerialNum);
 
   //Flashable partition running on FPGA
-  // the vectors are being populated by empty strings which need to be removed
-  auto remove_empty_strings = [](std::vector<std::string> uuids) {
-    auto it = uuids.begin();
-    while(it != uuids.end()) {
-      if((*it).empty())
-        uuids.erase(it);
-      else
-        it++;
-    }
-  };
-  
+
   std::vector<std::string> logic_uuids, interface_uuids;
+  // the vectors are being populated by empty strings which need to be removed
   try {
     logic_uuids = xrt_core::device_query<xrt_core::query::logic_uuids>(device);
-    remove_empty_strings(logic_uuids);
+    logic_uuids.erase(
+      std::remove_if(logic_uuids.begin(), logic_uuids.end(),	
+                      [](const std::string& s) { return s.empty(); }));
   } catch (...) {}
   try {
     interface_uuids = xrt_core::device_query<xrt_core::query::interface_uuids>(device);
-    remove_empty_strings(interface_uuids);
+    interface_uuids.erase(
+      std::remove_if(interface_uuids.begin(), interface_uuids.end(),	
+                  [](const std::string& s) { return s.empty(); }));

```

Comment:
This erases one element only and the other moved elements are bogus.  Make sure to add the end() iterator also. See https://en.cppreference.com/w/cpp/container/vector/erase.

std::remove_if returns an iterator to the new logic end of the vector, erase should erase the range [itr, end[ where itr is the new logical end returned by remove_if and end is the end() of the vector itself.
`vec.erase(std::remove_if(b,e,cond), vec.end());`

Reply:
Oh, that is what was missing from my initial implementation. Thanks for the clarification. I couldn't figure out why my logic wasn't working earlier. It was missing the end() :)